File: opaia_backend/app/models/car.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.db.base_class import Base
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from app.models.user import User
    from app.models.car_image import CarImage
    from app.models.rent import Rent

class Car(Base):
    __tablename__ = "cars"

    id = Column(Integer, primary_key=True, index=True)
    marca = Column(String, nullable=False)
    modelo = Column(String, nullable=False)
    ano = Column(Integer, nullable=False)
    preco = Column(Float, nullable=True)
    quilometragem = Column(Integer, nullable=True)
    descricao = Column(Text, nullable=True)

    dono_id = Column(Integer, ForeignKey("users.id"))
    dono = relationship("User", back_populates="cars")

    images = relationship("CarImage", back_populates="car", cascade="all, delete-orphan")
    rents = relationship("Rent", back_populates="car")

File: opaia_motors_frontend/lib/models/car.dart
class Car {
  final int id;
  final String marca;
  final String modelo;
  final int ano;
  final double preco;
  final String descricao;
  final String? imagemUrl;
  final int donoId;

  Car({
    required this.id,
    required this.marca,
    required this.modelo,
    required this.ano,
    required this.preco,
    required this.descricao,
    this.imagemUrl,
    required this.donoId,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'],
      marca: json['marca'],
      modelo: json['modelo'],
      ano: json['ano'],
      preco: json['preco'].toDouble(),
      descricao: json['descricao'],
      imagemUrl: json['imagem_url'],
      donoId: json['dono_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'marca': marca,
      'modelo': modelo,
      'ano': ano,
      'preco': preco,
      'descricao': descricao,
      'imagem_url': imagemUrl,
      'dono_id': donoId,
    };
  }
}
